### PR TITLE
Reset to auto if not set

### LINKED ISSUE #19109: -1 is not a valid value for maximal versions retention obligation
When uploading files to Files
`"method":"POST","url":"\/owncloud\/index.php\/apps\/files\/ajax\/upload.php`

This message shows up:
`"files_versions","message":"-1 is not a valid value for maximal versions retention obligation. Check versions_retention_obligation in your config.php. Falling back to auto.`

The default config contains 'auto', so there must be a check somewhere which uses '-1'.

### PR DESCRIPTION
**Pull request by VicDeo**
Fixes https://github.com/owncloud/core/issues/19109

### CODE REVIEW

**Comment by VicDeo** [comment]
@LukasReschke

**Comment by MorrisJobke** [comment]
Error message is gone :+1:

**Comment by MorrisJobke** [comment]
cc @oparoz

File: apps/files_versions/lib/expiration.php [hunk 1]
```diff
@@ -138,7 +138,7 @@ private function parseRetentionObligation(){
 		}
 
 		if (!isset($splitValues[1])) {
-			$maxValue = self::NO_OBLIGATION;
```
Comment (DeepDiver1975) [inline apps/files_versions/lib/expiration.php]: Drop NO_OBLIGATION since it's an invalid value????

Comment (VicDeo) [inline apps/files_versions/lib/expiration.php]: @DeepDiver1975 it's invalid for a local variable (that is set by splitting a value from config.php), not for minAge/maxAge class members below

Comment (VicDeo) [inline apps/files_versions/lib/expiration.php]: a value in config.php can be `auto`, `auto,auto`, `auto,_digit_`, `_digit_, auto`, `digit, digit`

this just is an intermediate step before parsing this value into `Expiration::minAge` and `Expiration::maxAge` that are not polymorphic but integer. `NO_OBLIGATION` is `-1` (or always evaluated as `tomorrow` relative to the current time)

**Comment by rullzer** [comment]
How hard would it be to have a unit tests for this?

File: apps/files_versions/lib/expiration.php (hunk 1 above)
Comment (DeepDiver1975) [inline apps/files_versions/lib/expiration.php]: THX @VicDeo

**Comment by VicDeo** [comment]
it's already covered and works as expected with/without this change.
The only difference is additional log entry

**Comment by rullzer** [comment]
> it's already covered and works as expected with/without this change.
> The only difference is additional log entry

OK :) :+1: then

**Comment by nickvergessen** [comment]
> The only difference is additional log entry

Well inject the logger instead of using OC::$server->getLogger() inside, and then check whether it was called...

**Comment by VicDeo** [comment]
@nickvergessen too much for such a minor thing.

**Comment by LukasReschke** [comment]
Works :+1:

**Pull request merged**